version.go

```diff
@@ -0,0 +1,100 @@
+package rindb
+
+import "fmt"
+
+// FileMeta holds metadata about an SSTable file.
+type FileMeta struct {
+	Number   uint64
+	Level    int
+	Smallest InternalKey
+	Largest  InternalKey
+	Size     uint64
+	SeqLo    uint64
+	SeqHi    uint64
+}
+
+// DeletedFileMeta references a file to remove.
+type DeletedFileMeta struct {
+	Level  int
+	Number uint64
+}
+
+// VersionEdit describes a change to the VersionSet.
+type VersionEdit struct {
+	ComparatorName string
+	LastSequence   uint64
+	NextFileNumber uint64
+	LogNumber      uint64
+	PrevLogNumber  uint64
+
+	AddFiles    []FileMeta
+	DeleteFiles []DeletedFileMeta
+}
+
+// VersionSet represents the in-memory state of levels and file numbering.
+type VersionSet struct {
+	Levels         [][]FileMeta
+	Comparator     string
+	NextFileNumber uint64
+	LogNumber      uint64
+	PrevLogNumber  uint64
+	LastSequence   uint64
+}
+
+func (vs *VersionSet) ensureLevel(level int) {
+	for len(vs.Levels) <= level {
+		vs.Levels = append(vs.Levels, nil)
+	}
+}
+
+// coalesceNonZero returns newVal if it's not the zero value for its type,
+// otherwise returns existing.
+func coalesceNonZero[T comparable](existing, newVal T) T {
+	var zero T
+	if newVal != zero {
+		return newVal
+	}
+	return existing
+}
```

Comment:
```suggestion
func coalesceNonZero[T comparable](values ...T) T {
	
	return cmp.Or(values...)
}
```

Reply:
I woud keep the original version:
- no extra import just for this
- call order stays intuitive (`existing, newVal`)
- `values ...T` loses the distinction between old/new, original code is more idiomatic
- function is trivial and self-contained
- your change would also force updates to unit tests and existing call sites unnecessarily

Reply:
ah, i see. You can use like this
```suggestion
func coalesceNonZero[T comparable](existing, newVal T) T {
	return cmp.Or(newVal, existing)
}
```

Reply:
i don't think extra import is a problem. Use built-in functions for best practice

Reply:
Ah, it's just an optional for best practice. You can bypass this comment if you think it is not important